relay-sampling/Cargo.toml

```
@@ -9,14 +9,20 @@ edition = "2021"
 license-file = "../LICENSE"
 publish = false
 
+[features]
+default = []
+redis = ["dep:relay-redis"]
+
 [dependencies]
+anyhow = { workspace = true }
```

It looks like `anyhow` is only needed for the redis feature, so you can update this to:
```suggestion
redis = ["dep:anyhow", "dep:relay-redis"]

[dependencies]
anyhow = { workspace = true, optional = true }
```